Getting started interactive message example does not work.

### Description

Hey! 
First, thanks for the great project. 🚀 

I've tried the getting started interactive message example but I can't make it work so that the `button_click` listener is invoked. See the code here https://github.com/sfroestl/bolt-example/blob/master/src/app.js
I've configured both `Interactive Components` and `Events API` of my slack app to point to `/slack/events`.
The message listener works fine.

https://slack.dev/bolt/tutorial/getting-started

### What type of issue is this?
- [x] bug

### Requirements
* [x] I've read and understood the [Contributing guidelines](https://github.com/slackapi/bolt/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

---

### Bug Report

Filling out the following details about bugs will help us solve your issue sooner.

#### Reproducible code in:
https://github.com/sfroestl/bolt-example

package version: 1.0.0

node version: v10.15.1

OS version(s): MacOs 10.14.4

#### Steps to reproduce:

1. Clone repo https://github.com/sfroestl/bolt-example and `$ npm i && npm start` 
2. Enter slack and send the bot a message with "Hi".
3. Click button.

#### Expected result:

Callback to be executed.
```
app.action('button_click', ({ action, say }) => {
  say(`@{action.user} clicked the button`);
});
```

#### Actual result:

The handler is not executed

Thanks a lot for your help!

I've been able to reproduce this issue with a simple button block action. Here is my code:

```javascript
const { App } = require('@slack/bolt');

const app = new App({
  token: process.env.SLACK_BOT_TOKEN,
  signingSecret: process.env.SLACK_SIGNING_SECRET,
});

app.message('hello', ({ say }) => {
  say({
    blocks: [{
      type: 'section',
      text: {
        type: 'mrkdwn',
        text: 'You can add a button alongside text in your message. ',
      },
      accessory: {
        type: 'button',
        action_id: 'foo',
        text: {
          type: 'plain_text',
          text: 'Button',
          emoji: true,
        },
        value: 'click_me_123',
      },
    }],
  });
});

app.action('foo', ({ ack }) => {
  console.log('action listener called');
  ack();
});

(async () => {
  const server = await app.start(process.env.PORT || 3000);
  console.log('⚡️ Bolt app is running!', server.address());
})();
```

Here is the output:

```
⚡️ Bolt app is running! { address: '::', family: 'IPv6', port: 3000 }
[ERROR]   { Error: An incoming event was not acknowledged before the timeout. Ensure that the ack() argument is called in your listeners.
    at receiverAckTimeoutError (/Users/ankur/Developer/play/bolt-demos/first-release/node_modules/@slack/bolt/dist/ExpressReceiver.js:167:19)
    at Timeout.setTimeout [as _onTimeout] (/Users/ankur/Developer/play/bolt-demos/first-release/node_modules/@slack/bolt/dist/ExpressReceiver.js:39:32)
    at ontimeout (timers.js:436:11)
    at tryOnTimeout (timers.js:300:5)
    at listOnTimeout (timers.js:263:5)
    at Timer.processTimers (timers.js:223:10) code: 'slack_bolt_receiver_ack_timeout_error' }
```

Notice that the `action listener called` output never appears.

---

We had an issue with block action (or interactive message) detection. Specifically, the incoming action (potentially a block action, interactive message, dialog submission, or message action) is supposed to go through the following function, which is supposed to detect whether its of the two types we're trying to identify. In this case, this detection gives us the wrong answer.

https://github.com/slackapi/bolt/blob/1ed27bbfbf2bd196723aafeccdf299e6b26f4bac/src/App.ts#L481-L485

The `action` property that its checking should be `actions`.

This is the kind of issue that TypeScript's typechecker is supposed to help us ensure that it doesn't happen. It's the reason we went through a lot of effort to write types for all these objects. However, in this case we defeated ourselves by leaving many objects "open ended", using the `StringIndexed` interface in places where we weren't sure if there would be additional properties added later.

I'm starting to think that we should remove `StringIndexed` from usages that make it easy to shoot ourselves in the foot (like this one). I've got a PR to share shortly.

--- a/src/App.ts
+++ b/src/App.ts
@@ -481,7 +481,7 @@ function buildSource(
 function isBlockActionOrInteractiveMessageBody(
   body: SlackActionMiddlewareArgs['body'],
 ): body is SlackActionMiddlewareArgs<BlockAction | InteractiveMessage>['body'] {
-  return (body as SlackActionMiddlewareArgs<BlockAction | InteractiveMessage>['body']).action !== undefined;
+  return (body as SlackActionMiddlewareArgs<BlockAction | InteractiveMessage>['body']).actions !== undefined;
 }
 
 function defaultErrorHandler(logger: Logger): ErrorHandler {

--- a/src/types/actions/block-action.ts
+++ b/src/types/actions/block-action.ts
@@ -118,7 +118,7 @@ export interface DatepickerAction extends BasicElementAction<'datepicker'> {
  *
  * This describes the entire JSON-encoded body of a request from Slack's Block Kit interactive components.
  */
-export interface BlockAction<ElementAction extends BasicElementAction = BlockElementAction> extends StringIndexed {
+export interface BlockAction<ElementAction extends BasicElementAction = BlockElementAction> {
   type: 'block_actions';
   actions: [ElementAction];
   team: {
@@ -166,9 +166,3 @@ export type BlockChannelsSelectAction = BlockAction<ChannelsSelectAction>;
 export type BlockExternalSelectAction = BlockAction<ExternalSelectAction>;
 export type BlockOverflowAction = BlockAction<OverflowAction>;
 export type BlockDatepickerAction = BlockAction<DatepickerAction>;
-
-// Relics of the past
-
-// type KnownActionFromElementType<T extends string> = Extract<ElementAction, { type: T }>;
-// type ActionFromElementType<T extends string> = KnownActionFromElementType<T> extends never ?
-//   BasicElementAction<T> : KnownActionFromElementType<T>;

--- a/src/types/actions/dialog-action.ts
+++ b/src/types/actions/dialog-action.ts
@@ -1,11 +1,9 @@
-import { StringIndexed } from '../helpers';
-
 /**
  * A Slack dialog submit action wrapped in the standard metadata.
  *
  * This describes the entire JSON-encoded body of a request from Slack dialogs.
  */
-export interface DialogSubmitAction extends StringIndexed {
+export interface DialogSubmitAction {
   type: 'dialog_submission';
   callback_id: string;
   submission: { [name: string]: string };

--- a/src/types/actions/interactive-message.ts
+++ b/src/types/actions/interactive-message.ts
@@ -1,5 +1,3 @@
-import { StringIndexed } from '../helpers';
-
 /**
  * All actions which Slack delivers from legacy interactive messages. The full body of these actions are represented
  * as [[InteractiveMessage]].
@@ -31,7 +29,7 @@ export interface MenuSelect {
  *
  * This describes the entire JSON-encoded body of a request from Slack's legacy interactive messages.
  */
-export interface InteractiveMessage<Action extends InteractiveAction = InteractiveAction> extends StringIndexed {
+export interface InteractiveMessage<Action extends InteractiveAction = InteractiveAction> {
   type: 'interactive_message';
   callback_id: string;
   actions: [Action];

--- a/src/types/actions/message-action.ts
+++ b/src/types/actions/message-action.ts
@@ -1,11 +1,9 @@
-import { StringIndexed } from '../helpers';
-
 /**
  * A Slack message action wrapped in the standard metadata.
  *
  * This describes the entire JSON-encoded body of a request from Slack message actions.
  */
-export interface MessageAction extends StringIndexed {
+export interface MessageAction {
   type: 'message_action';
   callback_id: string;
   trigger_id: string;

